src/components/Movies.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import Movie from './Movie';
+import FavMovie from './FavMovie';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+
+
+
+
+
+class Movies extends React.Component{
+  constructor(){
+    super();
+    this.state={
+      favList:[],
+      displayFav:false,
+
+
+    }
+    this.receiveFavClick = this.receiveFavClick.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.receiveDeleteClick = this.receiveDeleteClick.bind(this);
+  }
+
+  componentDidMount(){
+    const storageString = window.localStorage.getItem('favList');
+    const localFavList = !storageString ? []: JSON.parse(storageString)
+    this.setState({
+      favList: localFavList,
+    })
+  }
+
+  receiveFavClick(id){
+
+    if(this.state.favList.filter(movie => movie.imdbID==id).length == 0){
```

Comment:
It would be cleaner to calculate to generate a `favList` array in `if/else`, then you can set it in state and localStorage after and you would not need to duplicate code. Also, since `favList` would be precomputed we would not need to use a callback when setting it in localStorage